pkg/handler/enqueue_annotation.go

```diff
@@ -0,0 +1,173 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package handler
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var _ EventHandler = &EnqueueRequestForAnnotation{}
+
+const (
+	// NamespacedNameAnnotation defines the annotation that will be used to get the primary resource namespaced name.
+	// The handler will use this value to build the types.NamespacedName object used to enqueue a Request when an event
+	// to update, to create or to delete the observed object is raised. Note that if only one value be informed without
+	// the "/"  then it will be set as the name of the primary resource.
+	NamespacedNameAnnotation = "sigs.k8s.io/primary-resource-namespace"
+
+	// TypeAnnotation define the annotation that will be used to verify that the primary resource is the primary resource
+	// to use. It can be any arbitrary value. However, it would be recommended to fill it with `<Kind>.<group>` from
+	// the primary resource to, from a human-readability standpoint, make it easy to tell which specific resource
+	// is interested in. E.g Pods.core or ReplicaSet.apps.
+	TypeAnnotation = "sigs.k8s.io/primary-resource-type"
+)
+
+// EnqueueRequestForAnnotation enqueues Requests based on the presence of annotations that contain the type and
+// namespaced name of the primary resource. The purpose of this handler is to support cross-scope ownership
+// relationships that are not supported by native owner references.
+//
+// This handler should ALWAYS be paired with a finalizer on the primary resource. While the
+// annotation-based watch handler does not have the same scope restrictions that owner references
+// do, they also do not have the garbage collection guarantees that owner references do. Therefore,
+// if the reconciler of a primary resource creates a child resource across scopes not supported by
+// owner references, it is up to the reconciler to clean up that child resource.
+//
+// **NOTE** You might prefer to use the EnqueueRequestsFromMapFunc with predicates instead of it.
+//
+// **Examples:**
+//
+// The following code will enqueue a Request to the `primary-resource-namespace` when some change occurs in a Pod
+// resource:
+//
+// ...
+// if err := c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForAnnotation{"Pods.core"}); err != nil {
+//    entryLog.Error(err, "unable to watch Pods")
+//    os.Exit(1)
+// }
+// ...
+//
+// With the annotations:
+//
+// ...
+// annotations:
+//    sigs.k8s.io/primary-resource-name: "my-namespace/my-pod"
```

Comment:
godoc doesn't match the annotation name.

Reply:
solved